Kuadrant/limitador-operator
# Hello helm

This PR introduces a way to manage a Limitador Operator [Helm Chart](https://helm.sh/). This is not meant to replace the way we are building and delivering our manifests (Kustomize, OLM) but to provide an alternative (complementary) way of delivering the operator.

This early implementation uses Kustomize to create the chart template, instead of creating and maintaining new ones with Helm, to later customize the Helm only settings with its `values.yaml`

**NOTES**

* This chart (and every other operator chart we package in the future: Authorino, DNS, etc) bundles only the operator in the same fashion OLM does.
* The operator version matches the chart version
* This first release doesn't use helm templating, it provides only the manifests previously built by Kustomize.
* The sole repository for all our operator charts will be hosted in [Kuadrant/helm-charts](https://github.com/Kuadrant/helm-charts/).
* The "source of truth" (source code and releasing logic) will be provided in each operator repo, not in the helm charts repository above.
* The release workflow is not fully automated yet, just exposing `workflow_dispatch` it will change in the near future.
* The PR related for the repository is [WIP](https://github.com/Kuadrant/helm-charts/pull/8)
* Documentation will be provided once the repository and other operator charts are tested.
* The steps to try out the repository will be provided in the kuadrant helm charts repo.


**TODO**

* [x] Build manifests with Kustomize
* [x] Install/Uninstall and Upgrade targets
* [x] Adjust Charts mandatory settings
* [x] Decide on chart versioning (if matching operators or keep its own) ---> Same as OLM, same as Operator image
* [x] Github workflow for release package


**Verification Steps**

1. Create a local kind cluster
```sh
make kind-create-cluster
```

2. Deploy (install) the default (`latest`) Limitador Operator chart
```sh
make helm-install
```

3. Verify the installed operator image:
```sh
kubectl get deployments/limitador-operator-controller-manager -n limitador-operator-system -o yaml | grep image:
```

it should return:
```sh
image: quay.io/kuadrant/limitador-operator:latest
```

4. Build new manifests with specific version of the operator
```sh
make helm-build \
            VERSION=0.9.0 \
            LIMITADOR_VERSION=1.4.0
```

5. Upgrade the current installed operator to the freshly built one
```sh
make helm-upgrade
```

6. Check the installed operator image:
```sh
kubectl get deployments/limitador-operator-controller-manager -n limitador-operator-system -o yaml | grep image:
```

it should return:
```sh
image: quay.io/kuadrant/limitador-operator:v0.9.0
```
